GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -330,21 +352,77 @@ - (instancetype)init {
   return self;
 }
 
+/** Populates a unique NSUInteger as GDTCORBackgroundIdentifier for given semaphore and maps it to
+ * the semaphore in NSMutableDictionary.
+ *
+ * @param semaphore The task semaphore needs to be mapping.
+ * @return A unique GDTCORBackgroundIdentifier mapped to the given semaphore.
+ */
++ (GDTCORBackgroundIdentifier)populateBgIDwithSemaphore:(dispatch_semaphore_t)semaphore {
+  __block GDTCORBackgroundIdentifier bgID = GDTCORBackgroundIdentifierInvalid;
+  if (sSemaphoreQueue && sBgIDToSemephoreMap) {
+    dispatch_sync(sSemaphoreQueue, ^{
+      bgID = arc4random();
+      while (bgID == GDTCORBackgroundIdentifierInvalid ||
+             sBgIDToSemephoreMap[[NSNumber numberWithLong:bgID]]) {
+        bgID = arc4random();
+      }
+      sBgIDToSemephoreMap[[NSNumber numberWithLong:bgID]] = semaphore;
+    });
+  }
+  return bgID;
+}
+
+/** Returns the semaphore mapped be given bgID and removed the key value pair in
+ * NSMutableDictionary.
+ *
+ * @param bgID The unique NSUInteger as GDTCORBackgroundIdentifier.
+ * @return The semaphore mapped by given bgID.
+ */
++ (dispatch_semaphore_t)semaphoreForBgID:(GDTCORBackgroundIdentifier)bgID {
```

Comment:
maybe rename to semaphoreForBackgroundIdentifier

Reply:
Renamed